### PR TITLE
Add StateMixin utility functionality

### LINKED ISSUE #147: Create a Toolkit.MixinState Api utility
I'm still on the fence about this.

All this function would be is:
```
function(classDefinition) {
  _.extend(classDefinition.prototype, StateMixin);
}
```
However prototypical stuff scares people sometimes so this makes an easy interface that doesn't look scary.

But it is similar to https://github.com/marionettejs/backbone-metal/blob/master/src/mixin.js#L12 in its simplicity.

Alternatively people can always do this..  `const MyView = Marionette.ItemView.extend(StateMixin).extend({ foo: 'bar' });`  This is a perfectly acceptable method as well though it deepens the prototype chain.  Not sure what would be best to support.

### PR DESCRIPTION
**Pull request by TerrenceLJones**
Also move noconflict spec to new toolkit spec.

This resolves #147.

### CODE REVIEW

File: src/marionette.toolkit.js [hunk 1]
```diff
@@ -17,6 +17,16 @@ Toolkit.noConflict = function() {
   return this;
 };
 
+Toolkit.MixinState = function(classDefinition) {
+  let stateMixin = StateMixin;
```
Comment (paulfalgout) [inline src/marionette.toolkit.js]: so nit picky, but I just had a thought. lowerCase tends to indicate an instance of something.  What do you think about changing this var to `_StateMixin` or something like that?

File: docs/mixins/state.md [hunk 2]
```diff
@@ -31,6 +31,23 @@ var myClass = Marionette.Object.extend({
 
 _.extend(myClass.prototype, Marrionette.Toolkit.StateMixin)
 ```
+
+You can also use `Marionette.Toolkit.MixinState` which is a utility to mixin the `StateMixin` into any Marionette.Object or Marionette.View. If there is no `StateModel` definition on your class then the `StateModel` will be defined as a vanilla `Backbone.Model`. However, if you have already defined `StateModel` on your class, your `StateModel` definition **will not be overwritten**.
```
Comment (paulfalgout) [inline docs/mixins/state.md]: can we add  ` 's to Marionette.Object and Marionette.View ?

File: src/marionette.toolkit.js (hunk 1 above)
Comment (TerrenceLJones) [inline src/marionette.toolkit.js]: That sounds good...at the time I just couldn't think of a better name.

**Comment by TerrenceLJones** [comment]
@paulfalgout Should be good now.

**Pull request merged**